Examples/Cocoa/Sources/Objective_C/Pin.m

```diff
@@ -516,6 +527,132 @@ - (instancetype)mergeWithModel:(Pin *)modelObject initType:(PlankModelInitType)i
     [builder mergeWithModel:modelObject];
     return [[Pin alloc] initWithBuilder:builder initType:initType];
 }
+- (NSDictionary *)dictionaryObjectRepresentation
+{
+    NSMutableDictionary *dict = [[NSMutableDictionary alloc] initWithCapacity:16];
+    if (_pinDirtyProperties.PinDirtyPropertyAttribution) {
+        if (_attribution != nil) {
+            [dict setObject:_attribution forKey:@"attribution"];
+        } else {
+            [dict setObject:[NSNull null] forKey:@"attribution"];
+        }
+    }
+    if (_pinDirtyProperties.PinDirtyPropertyAttributionObjects) {
+        NSArray *items0 = _attributionObjects;
+        NSMutableArray *result0 = [NSMutableArray arrayWithCapacity:items0.count];
+        for (id obj0 in items0) {
+            if (obj0 != (id)kCFNull) {
+                [result0 addObject:[obj0 dictionaryObjectRepresentation]];
+            }
+        }
+        [dict setObject:result0 forKey:@"attribution_objects"];
+    }
+    if (_pinDirtyProperties.PinDirtyPropertyBoard) {
+        if (_board != nil) {
+            [dict setObject:[_board dictionaryObjectRepresentation] forKey:@"board"];
+        } else {
+            [dict setObject:[NSNull null] forKey:@"board"];
+        }
+    }
+    if (_pinDirtyProperties.PinDirtyPropertyColor) {
+        if (_color != nil) {
+            [dict setObject:_color forKey:@"color"];
+        } else {
+            [dict setObject:[NSNull null] forKey:@"color"];
+        }
+    }
+    if (_pinDirtyProperties.PinDirtyPropertyCounts) {
+        if (_counts != nil) {
+            [dict setObject:_counts forKey:@"counts"];
+        } else {
+            [dict setObject:[NSNull null] forKey:@"counts"];
+        }
+    }
+    if (_pinDirtyProperties.PinDirtyPropertyCreatedAt) {
+        NSValueTransformer *valueTransformer = [NSValueTransformer valueTransformerForName:kPlankDateValueTransformerKey];
```

Comment:
Probably a `nil` check here as well rather than relying on the date transformer to do this